dataGenerator/itemData.js

```diff
@@ -0,0 +1,161 @@
+const faker = require('faker');
+const fs = require('fs');
+
+const getBooleans = (product) => { // these are just 50/50 true/false
+  product.free_returns = (!Math.round(Math.random()));
+  product.easy_exchange = (!Math.round(Math.random()));
+  product.free_shipping = (!Math.round(Math.random()));
+  product.two_day_shipping = (!Math.round(Math.random()));
+};
+
+const getShipDays = (product) => {
+  const range = (Math.floor(Math.random() * 3)); // 1/3rd of the time, one of these options
+  if (range === 0) { product.ship_days = 2; } // 2
+  if (range === 1) { product.ship_days = Math.floor(Math.random() * 5) + 2; } // between 2-7
+  if (range === 2) { product.ship_days = Math.floor(Math.random() * 12) + 2; } //  2-14
+};
+
+const getBoughtCount = (product) => {
+  const range = (Math.floor(Math.random() * 6)); // 1/5th of the time, one of these options
+  if (range === 0) { product.bought = 0; } // 0
+  if (range === 1) { product.bought = Math.floor(Math.random() * 11); } // between 0-10
+  if (range === 2) { product.bought = Math.floor(Math.random() * 91 + 10); } //  10-100
+  if (range === 3) { product.bought = Math.floor(Math.random() * 901 + 100); } //  100-1000
+  if (range === 4) { product.bought = Math.floor(Math.random() * 3001 + 1000); } //  1000-4000
+  if (range === 5) { product.bought = Math.floor(Math.random() * 30001 + 4000); } //  4000-34000
+};
+
+const getReviews = (product) => {
+  const range = (Math.floor(Math.floor() * 3));
+  product.reviews = (Math.floor(Math.random() * 101));
+  product.rating = (Math.floor(Math.random() * 11) / 2);
+};
+
+const getDateAndDiscount = (product, seed) => {
+  if (!Math.round(Math.random())) {
+    product.sale_price = [1, 2, 5, 10, 20][seed];
+    product.sale_expires = Math.floor(Math.random() * 3); // days remaining (sales end@midnight)
+    product.expires = product.sale_expires; // if discounted, sale ends when discount does
+  } else {
+    product.expires = Math.floor(Math.random() * 12) + 1; // otherwise, ends in 1-12 days
+  }
+};
+
+const getPrices = (product, seed) => { // retail prices end in .99; discounts end in .97
+  let priceSeed;
+  if (seed === 0) { // 20% retail:$3.99-$12.99
+    priceSeed = Math.floor(Math.random() * 10);
+    product.price_retail = (priceSeed + 3.99).toFixed(2);
```

Comment:
Because `toFixed` returns a string, `product.price_retail` is going to be string, not an integer. I'm not sure if that's what you intended.